### DISCUSSION TITLE
Automatically close the "Submit to Frigate+" modal once choice is made

### DISCUSSION THREAD

**Discussion by Mincka** [blakeblackshear/frigate#8435] [Ideas]
Would it be possible to close the modal once we click on "This is a X" or "This is not a X"?
It would be faster to submit images to Frigate+.
The indication are clear enough and if there's a mistake, the image can still be deleted on Frigate+ to avoid using it in the model.

Currently, it's also a little buggy because when I click on "Close" for one image after its submission, I click on the next "Send to Frigate+" button, and once the upload of the previous image is completed, the modal close itself.

**Comment by NickM-27** [comment]
This is already how it works, but it waits until the upload is finished

**Reply by Mincka** [reply]
Oh I understand the behavior of the modal closing itself then...
Still, there's no point to wait for the upload to finish to close the modal since it's an async process.
It should be fine to just close the modal without waiting for the upload to finish. It's a "close" button anyway, not a "cancel" button.

**Reply by NickM-27** [reply]
Not necessarily, if an error occured while uploaded then the modal would show that. If it just closed immediately then received an error it would either not be handled or it would not be clear which image the error related to

**Reply by NickM-27** [reply]
It also leads to the buggy behavior that you are seeing

**Reply by Mincka** [reply]
Maybe an alternative could be to use the "EDIT IN FRIGATE+" link or even the button itself to state that the upload failed since it's already the button that says what's happening with "UPLOADING". It would not break the flow of submission and multiple uploads could happen in parallel. That's what's happening to me, I can easily have 2 or 3 uploads at the same time when I want to submit a batch of images.

**Reply by russelltomkins** [reply]
I'm with @Mincka on this, the UI flow forcing an extra "click" to close is ultra frustrating especially since the upload is already an async and I know I have thousands of clicking "close" ahead of me in the coming months as a new Frigate+ user.

Some ideas to help improve the Submit to Frigate+ screen to enable with batch submission efforts:
1. Add a "Previous/Next" Event link to allowing moving between events to upload
2. Add a "Default Action" dropdown to shift between "Submit and Close" and "Submit and Next event"

In regards to "failed submissions", I see we already have a filter (13.0 Beta 4) to show images already submitted to Frigate+ , why not add a "failed submission" filter so we can reprocess any failed uploads? And maybe a warning message on the "Model Creation" page in Frigate+ to check for failed submissions and verification prior to creation?

Thinking out loud ... is there value in uploading an image that is already matching at 90% or above? If there is no value, then maybe a "guided batch submission" flow can be created, that shows all events of type "Car <90%" and then moves onto all events of type "Person <90%", the same kind of flow we are already familiar with in regards to Face Tagging in photo apps.

**Reply by Mincka** [reply]
@russelltomkins : Vote +1 on the idea to maybe give it more visibility.
Maybe a temporary solution would be to use a ViolentMonkey script to close the modal when links are clicked.
If I manage to create it, I will post it there.
